### PR TITLE
Coding

### PR DESCRIPTION
**Pull request by konyan**
- Add New Project
- Edit Current Project
- Delete Current Project
- Add New Todo 
- Edit todo
- Delete Todo
- Design

### CODE REVIEW

File: src/js/todo.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    this.description = document.querySelector('#description');
+    this.dueDate = document.querySelector('#dueDate');
+    this.priority = document.querySelector('#priority');
+    this.project = document.querySelector('#project');
+    this.status = false;
+    this.idInput = document.querySelector('#todoId');
+    this.projectsDropDown = document.querySelector('.project-select');
+    this.todos = document.querySelector('.todos');
+    this.todoSubmit = document.querySelector('.add-todo');
+  }
```
Comment (maelfosso) [inline src/js/todo.js]: The `Tightly Coupled Objects` principle is not reflected.

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello @konyan @gitnyasha 
Nice UI/UX. Really awesome.
However, there are some issues I would like to show you.

1.
Screenshot_2020-06-05_09-47-11
There is that error occuring when submitting a TODO

2. The user must not be able to delete all the projects. There should always remain one project, the default project

3. Review your TODO class. Separate it from the DOM manipulation or extraction. Extract TODO data somewhere else and instantiate the class somewhere else.
Same thing with PROJECT.


Kindly make the changes and submit a new code review request.

**Review by mcrd25 — CHANGES_REQUESTED** [review]
# Changs Required

Hi Team!

I am currently unable to create a project or todo in the coding branch. Please address this issue!

[OPTIONAL] There is also a duplicate select statement that does nothing, consider removing it. 🙏 

Screen Shot 2020-06-06 at 11 44 14 AM

***Please makes changes and submit another review request.
If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,
Thank you for the changes done
However, there is still an error

1. Creating a TODO, Updating and Deleting it worked like a charm until I created a Project. When I wanted to create a TODO in this project I got this error in screenshot
Screenshot_2020-06-08_01-25-07
After an investigation, I think the problem come from the way you saved the new project in your localStorage.
Take a look at this screenshot
Screenshot_2020-06-08_01-32-14


2. You didn't take in consideration what is mentioned here https://github.com/konyan/todo-project/pull/1#discussion_r435790332
You should suppose that the model and controller can be used somewhere else without any modification. 
Your code cannot be used in such way because you strongly coupled the model and controller with the HTML section.
Extract data from HTML, then instantiate the Todo or Project and manipulate them. 

3. Please, don't forgot about the `default` project as indicated here https://github.com/konyan/todo-project/pull/1#pullrequestreview-425126514

Kindly make the changes and submit a new code review request
Thanks

Mael FOSSO

**Review by shloch — APPROVED** [review]
## Hello @konyan , @gitnyasha 

Nice job correcting the above bug
todo

I am following up with the former reviews especially the last one above this one (and the SECOND POINT), regarding tight coupling.

While this is not stated in the requirements, this is really good-practice advice, why not just follow even when it's emphasised a second time. Think about it.

What use would a javascript `Class` have class properties from an interface form-inputs ? This is not good programming at all. It's not professional to just ignore such a great advice.
It is also in the same light as former advice that asks that you separate LOGIC from UI. When you code this way, it's still not very much separated. This code is tide (coupled) to this interface.

You get a thumbs up, but consider what's said by TSEs. Discuss with them when you dont understand

### Happy coding

**Pull request merged**